src/main/java/com/aws/emr/profiler/cli/Config.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file.
+ * This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aws.emr.profiler.cli;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Optional;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.stream.JsonReader;
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.log4j.Logger;
+
+@Data
+@Builder
+public class Config {
+    private static Logger LOG = Logger.getLogger(Config.class);
+
+    @Getter
+    @Setter
+    private String s3Bucket;
+
+    @Getter
+    @Setter
+    private List<String> enabledProfilers;
+
+    public static Optional<Config> parseConfig(String configPath)  {
+        try {
+            File configFile = new File(configPath);
+            Gson serializer = new GsonBuilder().create();
+            return Optional.of(tryParseConfig(configFile, serializer));
+        } catch (Exception ex) {
+            LOG.error(ex);
+            return Optional.empty();
```

Comment:
Why are we swallowing exceptions here?

Reply:
We really aren't swallowing the exception because we throw an `InvalidConfigFileException`  if the `Optional<Config>` is empty in the main fn per...

```java
Config config = Config.parseConfig(DEFAULT_CONFIG_PATH)
.orElseThrow(InvalidConfigFileException::new);
```

I log the original exception for context for the customer. My assumption is it will usually be a `FileNotFoundException`.